libmscore/measure.cpp

```diff
@@ -3418,9 +3418,17 @@ qreal Measure::createEndBarLines(bool isLastMeasureInSystem)
             seg->setEnabled(false);
             }
       else {
-            BarLineType t = nm ? BarLineType::NORMAL : BarLineType::END;
+            BarLineType t = BarLineType::NORMAL;
             if (!seg)
                   seg = getSegmentR(SegmentType::EndBarLine, ticks());
+            else {
+                  //A segment exists, therefore we should continue this barline type
+                  BarLine* bl = toBarLine(seg->element(0));
+                  if (bl && bl->enabled())
```

Comment:
I think "enabled" is meaningful for the segment, not individual elements, unless barlines are different from other elements in this respect?